catalog/resource_metastore_data_access.go

```
@@ -115,6 +66,11 @@ func ResourceMetastoreDataAccess() common.Resource {
 						},
 					})
 					if err != nil {
+						// Rollback
+						rollback_err := acc.StorageCredentials.DeleteByMetastoreIdAndStorageCredentialName(ctx, metastoreId, dac.CredentialInfo.Name)
+						if rollback_err != nil {
+							return fmt.Errorf("failed to rollback: %v", rollback_err)
```

Here we're hiding original error - should we put it here as well?